src/pages/index.tsx

```diff
@@ -1,17 +1,54 @@
-import Head from 'next/head'
+import React, { useEffect, useState } from "react";
+import useSWR from "swr";
+import { signIn, useSession } from "next-auth/react"
+import { ICheckIn } from "@/models/checkins";
+
+let header = new Headers({ "Authorization": `Bearer placeholder-will-be-set` })
+let fetcher = (url: string) => fetch(url, { headers: header }).then((res) => res.json());
 
 export default function Home() {
-  return (
-    <>
-      <Head>
-        <title>Smart Events Viewer</title>
-        <meta name="description" content="Generated by create next app" />
-        <meta name="viewport" content="width=device-width, initial-scale=1" />
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
-      <main>
-
-      </main>
-    </>
-  )
-}
+
+  const { data: session, status } = useSession()
+
+  useEffect(() => {
+    header = new Headers({ "Authorization": `Bearer ${session?.id_token}` })
+    fetcher = (url: string) => fetch(url, { headers: header }).then((res) => res.json());
```

Comment:
Hm. I dislike the fetcher being stateful. Instead, try and make it a function that takes the header as a param. SWR should have a way of setting headers in the fetch.

Reply:
This was addressed in new commit. please see if this is better